pkg/common/configs/configvalidator.go

```diff
@@ -172,6 +172,11 @@ func checkLimitResource(cur QueueConfig, users map[string]map[string]*resources.
 				}
 				// Override with min resource
 				users[curQueuePath][user] = resources.ComponentWiseMinPermissive(limitMaxResources, userMaxResource)
+			} else if wildcardMaxResource, ok := users[queuePath][common.Wildcard]; user != common.Wildcard && ok {
+				if !wildcardMaxResource.FitInMaxUndef(limitMaxResources) {
+					return fmt.Errorf("user %s max resource %s of queue %s is greater than wildcard maximum resource %s of immediate or ancestor parent queue", user, limitMaxResources.String(), cur.Name, wildcardMaxResource.String())
+				}
+				users[curQueuePath][user] = limitMaxResources
```

Comment:
I think we would need to call `ComponentWiseMinPermissive` similar to earlier block before writing it into map. Otherwise, following case does not throw error. 

root: * => mem: 100
root.parent: user1=> mem: 50
root.parent.leaf: user2=> mem: 150

Reply:
The case is covered by test cases `leaf queue user maxresources exceed grandparent queue wildcard maxresources` and `leaf queue group maxresources exceed grandparent queue wildcard maxresources` in this PR. Yeah, we throw error for this config, because `user2` in `root.parent.leaf` is larger than `*` in `root`.

Reply:
Ok. I tried to recollect but not able to capture completely. May be because of error message mismatch in the above case, I would have that it has failed. Please try the below test code and you can see error message mismatch.

{
			name: "leaf queue user maxresources exceed grandparent queue wildcard maxresources",
			config: QueueConfig{
				Name: "parent",
				Limits: createLimitMaxResources(
					map[string]map[string]string{
						"*": {"memory": "50"},
					},
					nil),
				Queues: []QueueConfig{
					{
						Name: "child1",
						Limits: createLimitMaxResources(
							map[string]map[string]string{
								"user1": {"memory": "25"},
							},
							nil),
						Queues: []QueueConfig{
							{
								Name: "child2",
								Limits: createLimitMaxResources(
									map[string]map[string]string{
										"user2": {"memory": "75"},
									},
									nil),
							},
						},
					},
				},
			},
			hasError: true,
		},

Also, Are doing hierarchical comparison checks for wildcard user and group similar to named users and groups? I doubt this.

Reply:
Hi @manirajv06, thanks for the review. My bad, I changed `hasError` check to `errMsg` and forgot to remove the `hasError` attribute. That is why your test case is failed. May you change `hasError: true` to `errMsg: "is greater than wildcard maximum resource"`? I can pass with that. Thank you.

Reply:
Ok, Please go ahead.

 BTW, can you check the 2nd comment?

Reply:
> Are doing hierarchical comparison checks for wildcard user and group similar to named users and groups? I doubt this.

IMO, the validation is reasonable. If an admin gives limits like `user1: 100 vcores` and `*: 50 vcores` in the `root` queue, they may think user1 can use 100 vcores, but each of others can use only 50 vcores. In this situation, it's weird to set `user2: 100 vcores` in `root.child` queue.

Reply:
My question is,

root.parent => * => mem:10
root.parent.child1 => * => mem:15

Are we doing this wild card limit check between queues at different hierarchy? Ideally, we should not allow this.

Reply:
Agree. Add more test cases to cover it.

* leaf queue user wildcard maxresources are within immediate parent queue wildcard maxresources
* leaf queue user wildcard maxresources are within grandparent queue wildcard maxresources
* leaf queue user wildcard maxresources exceed immediate parent queue wildcard maxresources
* leaf queue user wildcard maxresources exceed grandparent queue wildcard maxresources